graphene/types/field.py

```diff
@@ -66,10 +68,39 @@ def __init__(
         self.deprecation_reason = deprecation_reason
         self.description = description
         self.default_value = default_value
+        self.permission_classes = permission_classes
+
+    def get_permissions(self):
+        """
+        Instantiates and returns the list of permissions that this field requires.
+        """
+        return [permission() for permission in self.permission_classes]
+
+    def check_permissions(self, info):
+        for permission in self.get_permissions():
+            if not permission.has_permission(info, self):
+                self.permission_denied(
+                    info, message=getattr(permission, 'message', None)
+                )
+
+    def permission_denied(self, info, message=None):
+        raise GraphQLError(message)
 
     @property
     def type(self):
         return get_type(self._type)
 
     def get_resolver(self, parent_resolver):
-        return self.resolver or parent_resolver
+        _resolver = self.resolver or parent_resolver
+
+        if not _resolver:
+            return None
+
+        def resolver(root, info, *args, **kwargs):
+
+            # TODO: pass root?
```

Comment:
Shouldn't kwargs  be passed as well?, I have cases where i need to check permissions based on input params

Reply:
maybe not it the same place but yes. You often times need the context (request) and all the inputs to validate. When you have inputs that depend on each other you need that. Although you shouldn't need inputs to determine field access, just to validate.

Reply:
agreed!

Reply:
Yup, I think we should pass all the arguments to the permission class, I was a bit lazy with the comment 😅